UnityProject/Assets/Scripts/Objects/FloorHazard.cs

```diff
@@ -0,0 +1,146 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using AddressableReferences;
+using HealthV2;
+using NaughtyAttributes;
+using UnityEngine;
+using UnityEngine.EventSystems;
+
+namespace Objects
+{
+	public class FloorHazard : IStepEvent, ICheckedInteractable<HandActivate>
+	{
+		[SerializeField] private AttackType attackType = AttackType.Melee;
+		[SerializeField] private DamageType damageType = DamageType.Brute;
+		[SerializeField, ShowIf("canCauseTrauma")] private TraumaticDamageTypes traumaType = TraumaticDamageTypes.NONE;
+		[SerializeField] private float damageToGive = 5f;
+		[SerializeField] private float armorPentration = 0f;
+		[SerializeField, ShowIf("canCauseTrauma")] private float traumaChance = 0f;
+		[SerializeField] private List<AddressableAudioSource> onStepSounds;
+		[SerializeField] private bool hurtsOneFootOnly;
+		[SerializeField] private bool ignoresFootwear;
+		[SerializeField] private bool ignoresHandwear;
+		[SerializeField] private bool isTrap;
+		[SerializeField] private bool canCauseTrauma;
+		[SerializeField, ShowIf("isTrap")] private bool isArmed;
```

Comment:
I really really dislike making objects change their identity based on a bool. That makes a lot of members of the class not used depending on the value of that bool. I would instead make ``FloorHazard`` and abstract class and extend it on ``MouseTrap`` and I guess the other one is glass shards?

Reply:
I have a plans for that in the next PR

Reply:
Reviews take a bit long so i decided to take that time and move the thing to it's own script instead of waiting until the next PR :P